src/docs/product/issues/issue-details/performance-issues/main-thread-io.mdx

```diff
@@ -1,30 +1,33 @@
 ---
-title: "IO on Main Thread for Mobile"
+title: "File IO on Main Thread"
 sidebar_order: 20
 description: "Learn more about File IO on Main Thread and how to diagnose and fix these issues."
 ---
 
 <Include name="early-adopter-note.mdx" />
 
-The Main UI thread in a mobile application handles user interface events such as button presses and page scrolls. If the Main UI thread performs long running operations like File IO, it will block the whole UI until the operation is complete, preventing the user from interacting with the app. If the Main UI thread is blocked for a long period of time, it can cause App Hangs and Application Not Responding errors.
+The main UI thread in a mobile application handles user interface events such as button presses and page scrolls. To prevent App Hangs and Application Not Responding errors, the main UI thread should not be used for performing long-running operations like File IO. These kinds of operations block the whole UI until they finish running and get in the way of the user interacting with the app.
```

Comment:
Hmm I think its important here we mention that IO on the main thread causes app hangs & ANR. Could we do something like

```suggestion
The main UI thread in a mobile application handles user interface events such as button presses and page scrolls. To prevent App Hangs and Application Not Responding errors, the main UI thread should not be used for performing long-running operations like File IO. These kinds of operations block the whole UI until they finish running and can lead to App Hangs and Application Not Responding errors.
```

Reply:
That's what we had before, but I changed it because I feel like we make it clear in the sentence above so this is kind of repetitive. But if you think it helps to say it twice then I will defer to your opinion :)